packages/utils-worker/tests/test-perf.ts

```diff
@@ -0,0 +1,310 @@
+    };
+};
+
+const generateHTML = (resources: Resource[] = []) => {
+    let result = `
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width">
+        <title>Basic Hints Test</title>`;
+
+    const css = resources.filter((res) => {
+        return res.type === ResourceType.CSS;
+    });
+
+    if (css.length) {
+        for (const { path } of css) {
+            result += `<link rel="stylesheet" href="${path}"/>`;
+        }
+    }
+
+    result += `
+    </head>
+    <body>
+        <h1>Basic Hints Test</h1>
+        <dialog open>
+            <form>`;
+
+    for (let i = 0; i < 500; i++) {
+        result += `<img src="icon.png"><button class="button">Test</button>
+                  `;
+    }
+
+    result += `</form>
+        </dialog>
+    </body>
+`;
+
+    const js = resources.filter((res) => {
+        return res.type === ResourceType.JS;
+    });
+
+    if (js.length) {
+        for (const { path } of js) {
+            result += `<script src="${path}"></script>`;
+        }
+    }
+
+    result += `
+</html>
+`;
+
+    return result;
+};
+
+const generateDeepHTML = (resources: Resource[] = []) => {
+    const deep = 100;
+    let result = `
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width">
+        <title>Basic Hints Test</title>`;
+
+    const css = resources.filter((res) => {
+        return res.type === ResourceType.CSS;
+    });
+
+    if (css.length) {
+        for (const { path } of css) {
+            result += `<link rel="stylesheet" href="${path}"/>`;
+        }
+    }
+
+    result += `
+    </head>
+    <body>
+        <h1>Basic Hints Test</h1>
+        <dialog open>
+            <form>`;
+
+    for (let j = 0; j < 5; j++) {
+        for (let i = 0; i < deep; i++) {
+            result += `<div>
+            <img src="icon.png"><button>Test</button>
+            `;
+        }
+
+        for (let i = 0; i < deep; i++) {
+            result += `</div>`;
+        }
+    }
+
+    result += `</form>
+        </dialog>
+    </body>
+`;
+
+    const js = resources.filter((res) => {
+        return res.type === ResourceType.JS;
+    });
+
+    if (js.length) {
+        for (const { path } of js) {
+            result += `<script src="${path}"></script>`;
+        }
+    }
+
+    result += `
+</html>
+`;
+
+    return result;
+};
+
+const prefixOrderCSS = [generateCSS('./index.css')];
+const leadingDotClasslistJS = [generateJS('./index.js')];
+
+const tests: Test[] = [{
+    expectedHints: ['axe/text-alternatives'],
+    expectedTime: 26000,
+    html: generateHTML(),
+    name: 'Axe perf test',
+    timeout: 40000
+}, {
+    expectedHints: ['axe/text-alternatives'],
+    expectedTime: 40000,
+    html: generateDeepHTML(),
+    name: 'Axe deep perf test',
+    timeout: 60000
+}, {
+    expectedHints: ['button-type'],
+    expectedTime: 3000,
+    hints: {
```

Comment:
Nit: since we're disabling all the axe tests multiple times, let's re-use a shared object so we don't have to repeat this. Even if we need to make slight modifications on top of it we can use the object-spread `{...obj}` syntax to blend them.